### PR TITLE
Telemetry2

### PR DESCRIPTION
**Pull request by mixedpuppy**
technical review first, we'll need to deal with data reviews before merging to main.

### CODE REVIEW

File: extension/api.js [hunk 1]
```diff
@@ -112,15 +112,27 @@ const lostEngines = new Map ([
 // we will try again later.
 let tryagain = false;
 
-function finish(result = false) {
-  Services.prefs.setBoolPref(RUN_ONCE_PREF, result);
+function finish(result = false, event, reason, id = null) {
```
Comment (Rob--W) [inline extension/api.js:115]: All parameters here appear to be mandatory, for this logic to make sense. Let's remove the default values.

**Review by Rob--W — CHANGES_REQUESTED** [review]

File: extension/api.js (hunk 1 above)
Comment (mixedpuppy) [reply extension/api.js:115]: id is only included if the panel is shown, so it will remain with a default.

File: extension/api.js [hunk 2]
```diff
@@ -32,18 +47,30 @@ const DEFAULT_SEARCH_SETTING_NAME = "defaultSearch";
 // If the value exists, we have ran before and will not run again.  If
 // we showed the panel, the value will be true.  We do not track what
 // the result of showing the panel is (ie. the user accepted or not).
```
Comment (wagnerand) [inline extension/api.js:49]: We do not track the user decision in the code (we track it via telemetry), but should we start doing so? If we need another iteration of this, we could bail out early if we know the previous user decision.

File: extension/api.js [hunk 3]
```diff
@@ -78,149 +105,243 @@ const lostEngines = new Map ([
...
-      // We only prompt for the first addon that makes it here.  If the user does
-      // not respond to the panel, they will be asked again on next startup.
+    } catch (err) {
+      // If the telemetry throws just log the error so it doesn't break any
+      // functionality.
+      Cu.reportError(err);
+    }
+
+    // We only prompt for the first addon that makes it here.  If the user does
+    // not respond to the panel, they will be asked again on next startup.
```
Comment (wagnerand) [inline extension/api.js:260]: Do we want to record in telemetry if the user did not respond?

File: extension/api.js [hunk 4]
```diff
@@ -78,149 +105,243 @@ const lostEngines = new Map ([
...
+  }
+  // If we made it here, no addon was currently eligible.  If any addons were
+  // blocklisted we will want to be able to prompt when it is released from
+  // the blocklist.
+  tryagain = tryagain || enabledAddons.length < addons.length;
+  if (!tryagain) {
+    console.log("reset-default-search: no enabled addons fit the criteria, finished.");
+    finish(false, "skipped", "noAddonsEligible");
+  }
+  console.log("reset-default-search: no enabled addons fit the criteria, waiting for more updates.");
```
Comment (wagnerand) [inline extension/api.js:271]: Do we want to record this in telemetry?

**Review by wagnerand — COMMENTED** [review]
General question: There are a few cases where we'd bail but try again later. Do we want to record those events in telemetry as well?

File: extension/api.js (hunk 3 above)
Comment (mixedpuppy) [reply extension/api.js:260]: We have no good way to determine that.  We do record that the panel is shown, then we record the response.  We'll see if there are more "asks" than responses in the data.

File: extension/api.js (hunk 4 above)
Comment (mixedpuppy) [reply extension/api.js]: added

File: extension/api.js (hunk 2 above)
Comment (mixedpuppy) [reply extension/api.js:49]: Would we actually show the panel again to a user who previously said no?  or yes?

**Review by Rob--W — APPROVED** [review]
Telemetry changes lgtm, but please squash and rebase the changes after merging #15 (the manifest version bump should probably be a separate commit).

File: extension/api.js (hunk 2 above)
Comment (wagnerand) [reply extension/api.js:49]: At this point, anything is possible. Storing the decision at least gives us the option not to prompt them again. We might not use it after all, but it's better to have it and not need it than he other way around.

File: extension/api.js [hunk 5]
```diff
@@ -190,7 +190,7 @@ async function tryDefaultReset() {
     let searchProvider = manifest?.chrome_settings_overrides?.search_provider;
     if (!searchProvider?.is_default) {
       // If the extension isn't asking to be default at this point, bail out.
-      console.log("reset-default-search: is_default is not requested by the addon, try again later");
+      console.log("reset-default-search: is_default is not requested by ${addon.id}, try again later");
```
Comment (wagnerand) [inline extension/api.js:193]: Reiterating previous comment (it's marked 'outdated' due to the other change): Is "Try again later" correct here? I don't see `tryagain` being set and the comment about says "bail out".

File: extension/manifest.json [hunk 6]
```diff
@@ -7,7 +7,7 @@
       "strict_min_version": "81.0"
     }
   },
-  "version": "1.0.6",
+  "version": "2.0.0",
```
Comment (kewisch) [inline extension/manifest.json:10]: Can you make sure to update package.json as well?

File: extension/api.js [hunk 7]
```diff
@@ -78,149 +105,257 @@ const lostEngines = new Map ([
...
+    console.log("reset-default-search: starting.");
+
     // We only run this once, after which we bail out.
     if (Services.prefs.prefHasUserValue(RUN_ONCE_PREF)) {
+      console.log("reset-default-search: has already ran once, exit.");
       return;
     }
+
     Services.telemetry.registerEvents("defaultSearchReset", {
+      skipped: {
```
Comment (kewisch) [inline extension/api.js:320]: Can you also update data.md accordingly to cover the telemetry events and parameters we're adding?

File: extension/api.js (hunk 2 above)
Comment (mixedpuppy) [reply extension/api.js:49]: Ok, the telemetry reason is now stored in the pref

**Comment by mixedpuppy** [comment]
I think I just did something wrong with git, the diff here doesn't seem right, but it does have the changes

**Review by wagnerand — APPROVED** [review]

**Pull request merged**